server/activityItem/model.ts

```diff
@@ -9,34 +9,44 @@ import {
 	Index,
 } from 'sequelize-typescript';
 import type { InferAttributes, InferCreationAttributes, CreationOptional } from 'sequelize';
-import { ActivityItemKind, ActivityItemPayload } from 'types';
+import { InsertableActivityItem, RecursiveAttributes } from 'types';
 
 @Table
-export class ActivityItem extends Model<
-	InferAttributes<ActivityItem>,
-	InferCreationAttributes<ActivityItem>
+// export class ActivityItem<Kind extends ActivityItemKind = ActivityItemKind> extends Model<
+// 	InferAttributes<ActivityItem<Kind>>,
+// 	InferCreationAttributes<ActivityItem<Kind>>
+// > {
+export class ActivityItem<T extends InsertableActivityItem = InsertableActivityItem> extends Model<
+	InferAttributes<ActivityItem<T>, { omit: Extract<keyof ActivityItem<T>, keyof T> }> & T,
+	InferCreationAttributes<ActivityItem<T>, { omit: Extract<keyof ActivityItem<T>, keyof T> }> & T
 > {
+	// this overrides the default Date type to be compatible with existing code
+	declare createdAt: CreationOptional<string>;
+	declare updatedAt: CreationOptional<string>;
+
+	public declare toJSON: <M extends Model>(this: M) => RecursiveAttributes<M>;
```

Comment:
I also override `toJSON` on every model, because the default version does not serialize the associations. 
This helper `recursiveAttributes` does do that.

I tried to do it differently as well, e.g. by just setting the type of the associations to e.g. `Attributes<Pub>` instead of `Pub`, but that caused TypeScript to throw a lot of `Type instantiation is possibly infinitely deep` errors, which ya hate to see